### PR TITLE
Deprecate inconsistent annotations and daprd options

### PR DESCRIPTION
**Pull request by pruthvidhodda**
# Description

Normalize daprd options and k8s annotations

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1717 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/injector/pod_patch.go [hunk 1]
```diff
@@ -274,7 +294,12 @@ func getMetricsPort(annotations map[string]string) int {
 }
 
 func getAppID(pod corev1.Pod) string {
-	return getStringAnnotationOrDefault(pod.Annotations, appIDKey, pod.GetName())
+	id := getStringAnnotationOrDefault(pod.Annotations, appIDKey, "")
+	if id != "" {
+		return id
+	}
+
```
Comment (youngbupark) [inline pkg/injector/pod_patch.go:301]: Please add TODO comment for all deprecated annotations. This way helps us to recognize it later.

Comment (pruthvidhodda) [reply pkg/injector/pod_patch.go:301]: Have added a comment [at the top](https://github.com/dapr/dapr/pull/1865/files#diff-7fadd8b154bfb3e50073e295c476aa6eR80) already, guess that should be sufficient?

Comment (youngbupark) [reply pkg/injector/pod_patch.go:301]: Ah ok

**Comment by youngbupark** [comment]
@pruthvidhodda Can you please update yaml in each test apps?
example:
https://github.com/dapr/dapr/blob/c0a20034200eb0814ea4aadd5f2e3dee89f69af5/tests/apps/actorapp/service.yaml#L38

**Comment by pruthvidhodda** [comment]
> @pruthvidhodda Can you please update yaml in each test apps?
> example:
> https://github.com/dapr/dapr/blob/c0a20034200eb0814ea4aadd5f2e3dee89f69af5/tests/apps/actorapp/service.yaml#L38

I remember updating them already, let me double check.

**Comment by pruthvidhodda** [comment]
All yamls are updated. @youngbupark good to merge?

File: pkg/runtime/cli.go [hunk 2]
```diff
@@ -30,19 +30,24 @@ func FromFlags() (*DaprRuntime, error) {
 	daprInternalGRPCPort := flag.String("dapr-internal-grpc-port", "", "gRPC port for the Dapr Internal API to listen on")
 	appPort := flag.String("app-port", "", "The port the application is listening on")
 	profilePort := flag.String("profile-port", fmt.Sprintf("%v", DefaultProfilePort), "The port for the profile server")
-	appProtocol := flag.String("protocol", string(HTTPProtocol), "Protocol for the application: grpc or http")
+	appProtocol := flag.String("app-protocol", "", "Protocol for the application: grpc or http")
```
Comment (yaron2) [inline pkg/runtime/cli.go:33]: This is a breaking change. Why did we remove the default being HTTP?

Comment (pruthvidhodda) [reply pkg/runtime/cli.go:33]: This isn't a breaking change. The default here is removed as we need logic to be backward compatible. Please look at [this section](https://github.com/dapr/dapr/pull/1865/files#diff-7fadd8b154bfb3e50073e295c476aa6eR283-R289) where I made HTTP to be default.

Comment (yaron2) [reply pkg/runtime/cli.go:33]: Defaults should come from `cli.go` so that they're visible and explicit.

Comment (pruthvidhodda) [reply pkg/runtime/cli.go]: Makes sense, changed it.

Comment (pruthvidhodda) [reply pkg/runtime/cli.go]: @yaron2 good to merge?

**Pull request merged**